File: clean.py
import re
import os

def process_file(filepath):
    with open(filepath, 'r', encoding='utf-8') as f:
        content = f.read()

    # Generic property removals
    content = re.sub(r'\s*tokopediaPrice(\?)?:\s*number;?', '', content)
    content = re.sub(r'\s*lazadaPrice(\?)?:\s*number;?', '', content)
    content = re.sub(r'\s*tiktokPrice(\?)?:\s*number;?', '', content)
    
    content = re.sub(r'\s*tokopediaLink(\?)?:\s*string;?', '', content)
    content = re.sub(r'\s*lazadaLink(\?)?:\s*string;?', '', content)
    content = re.sub(r'\s*tiktokLink(\?)?:\s*string;?', '', content)
    
    content = re.sub(r'\s*tokopediaUrl(\?)?:\s*string;?', '', content)
    content = re.sub(r'\s*lazadaUrl(\?)?:\s*string;?', '', content)
    content = re.sub(r'\s*tiktokUrl(\?)?:\s*string;?', '', content)
    
    content = re.sub(r'\s*tokopediaAvailable(\?)?:\s*boolean;?', '', content)
    content = re.sub(r'\s*lazadaAvailable(\?)?:\s*boolean;?', '', content)
    content = re.sub(r'\s*tiktokAvailable(\?)?:\s*boolean;?', '', content)
    
    # Specific object property removals (with trailing commas)
    content = re.sub(r'\s*tokopediaLink:\s*\'[^\']*\'\s*,?', '', content)
    content = re.sub(r'\s*lazadaLink:\s*\'[^\']*\'\s*,?', '', content)
    content = re.sub(r'\s*tiktokLink:\s*\'[^\']*\'\s*,?', '', content)
    
    content = re.sub(r'\s*tokopediaPrice:\s*\d+\s*,?', '', content)
    content = re.sub(r'\s*lazadaPrice:\s*\d+\s*,?', '', content)
    content = re.sub(r'\s*tiktokPrice:\s*\d+\s*,?', '', content)
    
    # Inline array variants
    content = re.sub(r',\s*tokopediaPrice:\s*\d+', '', content)
    content = re.sub(r',\s*lazadaPrice:\s*\d+', '', content)
    content = re.sub(r',\s*tiktokPrice:\s*\d+', '', content)
    content = re.sub(r',\s*tokopediaUrl:\s*"[^"]*"', '', content)
    content = re.sub(r',\s*lazadaUrl:\s*"[^"]*"', '', content)
    content = re.sub(r',\s*tiktokUrl:\s*"[^"]*"', '', content)

    with open(filepath, 'w', encoding='utf-8') as f:
        f.write(content)

process_file('src/data/products.ts')
print('Processed products.ts')

File: clean.js
const fs = require('fs');

function processFile(filepath) {
    let content = fs.readFileSync(filepath, 'utf-8');

    // Generic property removals
    content = content.replace(/\s*tokopediaPrice(\?)?:\s*number;?/g, '');
    content = content.replace(/\s*lazadaPrice(\?)?:\s*number;?/g, '');
    content = content.replace(/\s*tiktokPrice(\?)?:\s*number;?/g, '');
    
    content = content.replace(/\s*tokopediaLink(\?)?:\s*string;?/g, '');
    content = content.replace(/\s*lazadaLink(\?)?:\s*string;?/g, '');
    content = content.replace(/\s*tiktokLink(\?)?:\s*string;?/g, '');
    
    content = content.replace(/\s*tokopediaUrl(\?)?:\s*string;?/g, '');
    content = content.replace(/\s*lazadaUrl(\?)?:\s*string;?/g, '');
    content = content.replace(/\s*tiktokUrl(\?)?:\s*string;?/g, '');
    
    content = content.replace(/\s*tokopediaAvailable(\?)?:\s*boolean;?/g, '');
    content = content.replace(/\s*lazadaAvailable(\?)?:\s*boolean;?/g, '');
    content = content.replace(/\s*tiktokAvailable(\?)?:\s*boolean;?/g, '');
    
    // Specific object property removals (with trailing commas)
    content = content.replace(/\s*tokopediaLink:\s*'[^']*'\s*,?/g, '');
    content = content.replace(/\s*lazadaLink:\s*'[^']*'\s*,?/g, '');
    content = content.replace(/\s*tiktokLink:\s*'[^']*'\s*,?/g, '');
    
    content = content.replace(/\s*tokopediaPrice:\s*\d+\s*,?/g, '');
    content = content.replace(/\s*lazadaPrice:\s*\d+\s*,?/g, '');
    content = content.replace(/\s*tiktokPrice:\s*\d+\s*,?/g, '');
    
    // Inline array variants
    content = content.replace(/,\s*tokopediaPrice:\s*\d+/g, '');
    content = content.replace(/,\s*lazadaPrice:\s*\d+/g, '');
    content = content.replace(/,\s*tiktokPrice:\s*\d+/g, '');
    content = content.replace(/,\s*tokopediaUrl:\s*"[^"]*"/g, '');
    content = content.replace(/,\s*lazadaUrl:\s*"[^"]*"/g, '');
    content = content.replace(/,\s*tiktokUrl:\s*"[^"]*"/g, '');

    fs.writeFileSync(filepath, content, 'utf-8');
}

processFile('src/data/products.ts');
console.log('Processed products.ts');
